pkg/controller/vpc_nat_gateway.go

```diff
@@ -0,0 +1,818 @@
+		return fmt.Errorf("vpcNatEnabled='%s'", vpcNatEnabled)
+	}
+	c.vpcNatGwKeyMutex.Lock(natGwKey)
+	defer c.vpcNatGwKeyMutex.Unlock(natGwKey)
+	gw, err := c.vpcNatGatewayLister.Get(natGwKey)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+
+	pod, err := c.getNatGwPod(natGwKey)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+
+	var toBeDelEips, oldEips []*kubeovnv1.Eip
+	if eipAnnotation, ok := pod.Annotations[util.VpcEipsAnnotation]; ok {
+		if err := json.Unmarshal([]byte(eipAnnotation), &oldEips); err != nil {
+			klog.Errorf("%v", gw.Spec.Eips)
+			return err
+		}
+	}
+
+	for _, oldEip := range oldEips {
+		toBeDel := true
+		for _, newEip := range gw.Spec.Eips {
+			if oldEip.EipCIDR == newEip.EipCIDR {
+				toBeDel = false
+				break
+			}
+		}
+		if toBeDel {
+			toBeDelEips = append(toBeDelEips, oldEip)
+		}
+	}
+
+	if len(toBeDelEips) > 0 {
+		var delRules []string
+		for _, rule := range toBeDelEips {
+			delRules = append(delRules, rule.EipCIDR)
+		}
+		if err = c.execNatGwRules(pod, "eip-del", delRules); err != nil {
+			klog.Errorf("failed to exec nat gateway rule, err: %v", err)
+			return err
+		}
+	}
+
+	if len(gw.Spec.Eips) > 0 {
+		var addRules []string
+		for i, rule := range gw.Spec.Eips {
+			addRules = append(addRules, fmt.Sprintf("%s,%s,%d", rule.EipCIDR, rule.Gateway, 100+i))
+		}
+		if err = c.execNatGwRules(pod, "eip-add", addRules); err != nil {
+			return err
+		}
+	}
+
+	eipBytes, err := json.Marshal(gw.Spec.Eips)
+	if err != nil {
+		klog.Errorf("marshal eip annotation failed %v", err)
+		return err
+	}
+	pod.Annotations[util.VpcEipsAnnotation] = string(eipBytes)
+	if _, err := c.config.KubeClient.CoreV1().Pods(pod.Namespace).Patch(context.Background(), pod.Name, types.JSONPatchType, generatePatchPayload(pod.Annotations, "replace"), metav1.PatchOptions{}, ""); err != nil {
+		klog.Errorf("patch pod %s/%s failed %v", pod.Name, pod.Namespace, err)
+		return err
+	}
+	return nil
+}
+
+func (c *Controller) handleUpdateVpcFloatingIp(natGwKey string) error {
+	if vpcNatEnabled != "true" {
+		return fmt.Errorf("vpcNatEnabled='%s'", vpcNatEnabled)
+	}
+	c.vpcNatGwKeyMutex.Lock(natGwKey)
+	defer c.vpcNatGwKeyMutex.Unlock(natGwKey)
+	gw, err := c.vpcNatGatewayLister.Get(natGwKey)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+
+	pod, err := c.getNatGwPod(natGwKey)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return nil
+		}
+		return err
+	}
+
+	// check md5
+	newMd5 := fmt.Sprintf("%x", structhash.Md5(gw.Spec.FloatingIpRules, 1))
+	oldMd5 := pod.Annotations[util.VpcFloatingIpMd5Annotation]
+	if newMd5 == oldMd5 {
+		return nil
+	}
+
+	// update rules
+	var rules []string
+	for _, rule := range gw.Spec.FloatingIpRules {
+		rules = append(rules, fmt.Sprintf("%s,%s", rule.Eip, rule.InternalIp))
+	}
+	if err = c.execNatGwRules(pod, "flouting-ip-sync", rules); err != nil {
```

Comment:
`flouting` -> `floating`

Reply:
it's better to use const to  replace the strings here

Reply:
already fix